Yosemite/Yosemite/Stores/CardPresentPaymentStore.swift

```diff
@@ -392,48 +390,32 @@ private extension CardReaderConfigError {
 
 // MARK: Networking Methods
 private extension CardPresentPaymentStore {
-    /// Switch the store to use WCPay as the backend.
-    /// Does nothing if the store is already using WCPay.
-    /// WCPay is the initial default for the store.
+    /// Sets the store to use a given payment gateway
     ///
-    func useWCPayAsBackend() {
-        guard usingBackend != .wcpay else {
-            return
-        }
-
-        usingBackend = .wcpay
-    }
-
-    /// Switch the store to use Stripe as the backend.
-    /// Does nothing if the store is already using Stripe.
-    ///
-    func useStripeAsBackend() {
+    func use(paymentGatewayAccount: PaymentGatewayAccount) {
         guard allowStripeIPP else {
             DDLogError("useStripeAsBackend called when stripeExtensionInPersonPayments disabled")
             return
         }
 
-        guard usingBackend != .stripe else {
+        guard paymentGatewayAccount.isWCPay else {
+            usingBackend = .stripe
             return
         }
 
-        usingBackend = .stripe
+        usingBackend = .wcpay
     }
 
     /// Loads the account corresponding to the currently selected backend. Deletes the other (if it exists).
     ///
     func loadAccounts(siteID: Int64, onCompletion: @escaping (Result<Void, Error>) -> Void) {
-        switch usingBackend {
-        case .wcpay:
-            loadWCPayAccount(siteID: siteID, onCompletion: onCompletion)
-        case .stripe:
-            loadStripeAccount(siteID: siteID, onCompletion: onCompletion)
-        }
+        loadWCPayAccount(siteID: siteID, onCompletion: onCompletion)
+        loadStripeAccount(siteID: siteID, onCompletion: onCompletion)
     }
 
     func loadWCPayAccount(siteID: Int64, onCompletion: @escaping (Result<Void, Error>) -> Void) {
-        /// Delete any Stripe account present. There can be only one.
-        self.deleteStaleAccount(siteID: siteID, gatewayID: StripeAccount.gatewayID)
+        /// Delete any WCPay account present. There can be only one.
+        self.deleteStaleAccount(siteID: siteID, gatewayID: WCPayAccount.gatewayID)
```

Comment:
The logic was a bit mixed here. The method to load the WCPay account was removing all Stripe accounts, and the method to load the Stripe account we removing all WCPay accounts

Reply:
I did that on purpose to keep stale accounts from no-longer-active plugins out of our stores. I'll double check the logic.